core/src/tasks/record/export.ts

```diff
@@ -29,49 +33,30 @@ export class RecordExport extends RetryableTask {
         state: "processing",
         recordId: record.id,
       },
-      order: [["createdAt", "asc"]],
     });
 
-    try {
-      const oldGroupIds = imports[0]?.oldGroupIds;
-      const newGroupIds = imports[imports.length - 1]?.newGroupIds;
-
-      const oldGroups =
-        oldGroupIds && oldGroupIds.length > 0
-          ? await Group.findAll({
-              where: { id: { [Op.in]: oldGroupIds } },
-            })
-          : [];
-
-      const newGroups =
-        newGroupIds && newGroupIds.length > 0
-          ? await Group.findAll({
-              where: { id: { [Op.in]: newGroupIds } },
-            })
-          : [];
-
-      const destinations = await DestinationOps.relevantFor(
-        record,
-        oldGroups,
-        newGroups
-      );
+    const destinations = await DestinationOps.relevantFor(
+      record,
+      record.groupMembers.map((gm) => gm.group)
+    );
 
-      // check for explicit destinations to export to from each import
-      for (const i in imports) {
-        const _import = imports[i];
-        if (
-          _import.data?._meta?.destinationId &&
-          !destinations
-            .map((d) => d.id)
-            .includes(_import.data?._meta?.destinationId)
-        ) {
-          const destination = await Destination.scope(null).findOne({
-            where: { id: _import.data._meta.destinationId },
-          });
-          if (destination) destinations.push(destination);
-        }
+    // check for explicit destinations to export to from each import
+    for (const i in imports) {
```

Comment:
Nitpick: I think you can do `for import of imports` and reference it directly instead of `_import`